advanced_source/extend_dispatcher.rst

```diff
@@ -0,0 +1,295 @@
+  More details can be found in :ref:`register kernels for the new backend <register-kernel>`_.
+
+
+.. _register-kernel:
+
+Register kernels for the new backend
+------------------------------------
+
+To register your kernels to PyTorch dispatcher using the ``TORCH_LIBRARY_IMPL`` API
+described in `Registering a Dispatched Operator in C++ <dispatcher>`_.
+
+.. code-block:: cpp
+
+  TORCH_LIBRARY_IMPL(aten, PrivateUse1, m) {
+    m.impl(<op_schema_name1>, &customized_kernel1);
+    m.impl(<op_schema_name2>, &customized_kernel2);
+  }
+
+Now let's zoom in and what operator requires a kernel from customized backend and what's
+inside ``customized_kernel1`` exactly.
+
+PyTorch currently has more than 1600 operators and it’s still growing.  It’s unrealistic
+for backend extensions to keep up with this speed.  Even for native backends like CPU
+or CUDA, it often requires a lot of work to write dedicated kernels for every new op.
+
+Fortunately, the PyTorch dispatcher allows registration to *alias dispatch keys* which can
+register the same kernel to multiple backends.  And some kernels are written in a way that
+they naturally decompose to combination of several known operators. In other words, those
+kernels work for all backends (and potentially Autograd which we’ll talk about later)!
+
+In general PyTorch operators can be classified into two categories:
+
+* Ops that require registration: PyTorch native implementation for these ops is backend specific
+  and thus it’s required to provide a kernel for customized backend.  Otherwise calling such op
+  on the customized backend will error out.
+    * In ``RegistrationDeclarions.h`` these operators have ``dispatch=True and default=False``.
+
+
+* Registration is optional: backend extenders can skip registering to these ops without sacrificing any support.
+  However, if a backend extender wants to override the default kernel provided by PyTorch, they can still
+  optionally register their customized kernel to their backend and the dispatcher will use it for your backend only.
+    * In ``RegistrationDeclarations.h`` these operators have ``dispatch=False or default=True``.
+
+
+
+Autograd support for the new backend
+------------------------------------
+
+In general PyTorch tensor extensions live below PyTorch’s Autograd layer, which means PyTorch handles
+backward graph setup before dispatching to device kernels.  Because gradient formulas are often purely mathematical
+and thus are general for all backends, PyTorch often registers a kernel to alias dispatch key Autograd,
+which means it can be used by all backends.
+
+In some cases, PyTorch backward kernel implementations are also device specific so that they can squeeze out
+max performance out of each backend. For those operators you’ll see op_backward showing up in
+``RegistrationDeclarations.h`` as *required registration* as well.
+
+In other words, for most operators you don't have to worry about their derivative formulas,
```

Comment:
I'd move this paragraph and the code sample that follows up above the "In some cases", and change the "In other words, for most operators..." to "For these operators...". That way you're describing the simplest case first, from start to finish.

Then you can follow with "In some cases..." and I think I'd add a similarly simple code sample for a backwards implementation and registration there. (The registration is important to clarify which dispatch key they should be using to register `<op>_backward`)